### ISSUE TITLE
Design questions

### ISSUE THREAD

**Issue by joshmoore** [ome/omero-test-infra#1]
* [ ] Should this repo be merged into https://github.com/openmicroscopy/omero-test-daemon-c7-docker?
 * [x] Should this be renamed `omero-test-infra`?
 * [ ] Should `.omeroci/test-data` be run in a client docker or the server docker?
 * [ ] Should we use `travis-folds` everywhere?

TODO:
 * [ ] Make `Dockerfile` in lib-docker (et al) configuration for multiple build systems (mvn, gradle)
 * [ ] Rename `test-data` to `setup`

**Comment by sbesson** [comment]
Re [omero-test-daemon-c7-docker](https://github.com/openmicroscopy/omero-test-daemon-c7-docker), it would make sense to re-use it from a historical perspective, considering the former as version `0.0.1.` of what this is trying to achieve. I don't know if we'll end up doing `git rm && git add` though Should https://hub.docker.com/r/openmicroscopy/omero-test-daemon-c7/ be deprecated/dropped? If we are not reusing the repository, we should probably deprecate, archive and redirect it to the new testing infrastructure.

I would vote for keeping the `omero-*` convention when possible. `omero-test-infra` would work for me.

What are the pros and cons of running `omeroci/test-data` either on the client docker or server docker? Are there cases where we would like to configure in which location this should be run? In which case, we could probably make that configurable using the filename or more advanced strategies.

**Comment by jburel** [comment]
I will make sense to re-use if  https://github.com/openmicroscopy/omero-test-daemon-c7-docker if possible

for the name I agree with @sbesson we should keep omero- as a prefix.``omero-test-infra`` is a good as any and I don't have a better suggestion

**Comment by joshmoore** [comment]
I've renamed this repo to omero-test-infra and the following 4 PRs are green:

 * https://github.com/ome/omero-cli-render/pull/1
 * https://github.com/ome/rOMERO-gateway/pull/22
 * https://github.com/ome/scripts/pull/136
 * https://github.com/ome/omero-mapr/pull/23

I was hoping to extract the PyTest command and make use of https://github.com/openmicroscopy/openmicroscopy/pull/5585 but it's too difficult to consume the devspace OMERO.server build. (Here we need a decoupled `python-omero` or `python-omero-testlib`)

My proposal would be to merge all 4 of those, perform a release where appropriate, and then consider this version `0.1.0` of the new test infra. I'd don't see a straight-forward way to reuse `omero-test-daemon` but if it's worth it, I'll invest more time.

**Comment by joshmoore** [comment]
Few decisions that have been made over scattered conversations (open for veto):

 * [*] rename `test-omero` to `omero-test-infra`
 * [ ] rename `.omeroci` to `.ome`
 * [ ] rename `.omero` to `.ome/omero-server`
 * [ ] add an `.ome/meta.yml` file including the likes of:
    - tag prefix
    - type of application, e.g. "I'm a web plugin"
    - ...

**Comment by joshmoore** [comment]
Discussion with @jburel and @sbesson today:

 * We'll soon need `app-build` and perhaps `add-test` (if not also, `app-robot`, etc).
 * Before we start adding such phases, propose to tag this as `0.1.0` including any final naming changes that are desired so we can begin iterating.

**Comment by jburel** [comment]
app-build already exists. it is currently used to run the tests
This might need to be renamed

**Comment by jburel** [comment]
I had to make various modification to use for figure and iviewer
(more adjustments due to iviewer repo layout)
I will hopefully have a PR tomorrow so we can start a discussion

**Comment by jburel** [comment]
Changes to have iviewer and figure work with test-infra
https://github.com/openmicroscopy/omero-test-infra/pull/3

**Comment by jburel** [comment]
Current implementation assumes that the scripts to test are in a folder ``omero``
we will need to relax to handle case like ``gs/metadata.py``
I had to make an change to work with figure but we should consider to go a step further

**Comment by jburel** [comment]
Changes have been made in #3 to handle scripts not directly under ``/omero``
Looking at other repositories e.g. https://github.com/openmicroscopy/omero-example-scripts
Ignoring the fact that it is not pip installable, there is a "flat" structure. This won't handle by the current implementation or changes in #3

**Comment by jburel** [comment]
Consider more configuration options for apps
